packages/components/src/public/QueryModel/QueryModel.ts

```diff
@@ -428,7 +428,7 @@ export class QueryModel {
     }
 
     get hasSelections(): boolean {
-        return this.selections !== undefined;
+        return this.selections?.size > 0;
```

Comment:
Have you checked all usages of this to make sure that you're not breaking intended behavior? IIRC I purposely made this method check undefined and not > 0 because I was using it to determine if selections had been loaded. I'm not going to hold up the PR on this, but please make sure that the usages of this are still valid. e.g. some usages may be doing something if we have 0 selections, something different for > 0, and something different for undefined (since not all gridpanels have selections enabled).

Reply:
> Have you checked all usages of this to make sure that you're not breaking intended behavior? IIRC I purposely made this method check undefined and not > 0 because I was using it to determine if selections had been loaded. I'm not going to hold up the PR on this, but please make sure that the usages of this are still valid. e.g. some usages may be doing something if we have 0 selections, something different for > 0, and something different for undefined (since not all gridpanels have selections enabled).

You were right there was a case with selectAll that was broken with this change.  I went through and updated/validated all usages of QueryModel.hasSelections.  I tested and verified tests on core components page, sample manager and biologics.  @labkey-alan can you review this PR now that its a little more than the original change.